### ISSUE TITLE
Life Cycle Events (add columns-populated event)

### ISSUE THREAD

**Issue by dereklin** [angular-ui/ui-grid#3496]
Is there documentation on all the events in the ui grid life cycle?  At onRegisterApi, the gridApi is registered, but the columns are not yet populated.  Is there another event where all the columns are populated?

**Comment by PaulL1** [comment]
The columns are populated asynchronously.  Usually we use onRegisterApi + run a $interval.  

There is not currently documentation on all the events in the grid lifecycle, there is some documentation that is related to that in http://ui-grid.info/docs/#/api/design-rendering-cycle

**Comment by dereklin** [comment]
I created a directive that use ui-grid in the template.  I needed to create an external filter.  This filter is outside of the grid.  So I used what others have talked about here and hard set the filter term.  However, since the columns are not yet populated in onRegisterApi, I can't set the term there.  I ended up using $timeout in my directive's link function to set the filter term.  It would be great if there is onGridPopulated in addition to onRegisterApi.
